src/context/AppProvider/hooks/useSnapStatusHandler.tsx

```diff
@@ -0,0 +1,112 @@
+import { usePrevious, useToast } from '@chakra-ui/react'
+import { btcAssetId } from '@shapeshiftoss/caip'
+import { useQueryClient } from '@tanstack/react-query'
+import { useEffect } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { useSelector } from 'react-redux'
+import { useNavigate } from 'react-router'
+
+import { WalletActions } from '@/context/WalletProvider/actions'
+import { useIsSnapInstalled } from '@/hooks/useIsSnapInstalled/useIsSnapInstalled'
+import { useModal } from '@/hooks/useModal/useModal'
+import { useWallet } from '@/hooks/useWallet/useWallet'
+import { METAMASK_RDNS } from '@/lib/mipd'
+import { selectEnabledWalletAccountIds, selectWalletId } from '@/state/slices/common-selectors'
+import { selectWalletRdns } from '@/state/slices/localWalletSlice/selectors'
+import { portfolio } from '@/state/slices/portfolioSlice/portfolioSlice'
+import { preferences } from '@/state/slices/preferencesSlice/preferencesSlice'
+import { useAppDispatch, useAppSelector } from '@/state/store'
+
+export const useSnapStatusHandler = () => {
+  const queryClient = useQueryClient()
+  const appDispatch = useAppDispatch()
+  const translate = useTranslate()
+  const toast = useToast()
+  const connectedRdns = useAppSelector(selectWalletRdns)
+  const previousConnectedRdns = usePrevious(connectedRdns)
+  const currentWalletId = useAppSelector(selectWalletId)
+  const walletAccountIds = useAppSelector(selectEnabledWalletAccountIds)
+  const { isSnapInstalled, isCorrectVersion } = useIsSnapInstalled()
+  const snapModal = useModal('snaps')
+  const previousSnapInstall = usePrevious(isSnapInstalled)
+  const previousIsCorrectVersion = usePrevious(isCorrectVersion)
+  const showSnapModal = useSelector(preferences.selectors.selectShowSnapsModal)
+  const {
+    state: { wallet, deviceId },
```

Comment:
also not using wallet in here

Reply:
Removed in [4bc0955](https://github.com/shapeshift/web/pull/9993/commits/4bc0955b005c8a33417d08be865579fe549409c6)